vortex-array/src/arrow/convert.rs

```diff
@@ -370,3 +385,35 @@ impl FromArrowArray<&dyn ArrowArray> for ArrayRef {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use arrow_array::{ArrayRef as ArrowArrayRef, StructArray as ArrowStructArray, new_null_array};
+    use arrow_buffer::NullBufferBuilder;
+    use arrow_schema::{DataType, Field, Fields};
+
+    use crate::ArrayRef;
+    use crate::arrow::FromArrowArray as _;
+
+    #[test]
+    pub fn nullable_may_contain_non_nullable() {
+        let fields = Fields::from(vec![Field::new(
+            "non_nullable_inner",
+            DataType::Int32,
+            false,
```

Comment:
I don't see why you're allowed to mark a field as non-nullable when it contains nulls?